core/src/main/scala/chisel3/experimental/hierarchy/core/Instance.scala

```diff
@@ -136,10 +153,11 @@ object Instance extends SourceInfoDoc {
   ): Instance[T] = {
     // Check to see if the module is already defined internally or externally
     val existingMod = Builder.definitions.view.map(_.proto).exists {
-      case c: Class               => c == definition.proto
-      case c: RawModule           => c == definition.proto
-      case c: BaseBlackBox        => c.name == definition.proto.name
-      case c: BaseIntrinsicModule => c.name == definition.proto.name
+      case c: Class                       => c == definition.proto
+      case c: RawModule                   => c == definition.proto
+      case c: ImportedDefinitionExtModule => c.importedDefinition == definition
+      case c: BaseBlackBox                => c.name == definition.proto.name
+      case c: BaseIntrinsicModule         => c.name == definition.proto.name
       case _ => false
```

Comment:
This is not a comment on this PR... but we need make this faster lol, this is a linear search on every single definition every time we make an instance 😑